src/plugins/GUITestBase/src/GTUtilsDashboard.cpp

```diff
@@ -24,17 +24,78 @@
 #include <primitives/GTWidget.h>
 
 #include <QRegularExpression>
+#include <QStringBuilder>
 #include <QTabWidget>
 
 #include <U2Gui/HoverQLabel.h>
 
 #include "GTUtilsDashboard.h"
 
+namespace {
+
+// A substring with the name of the notification type is searched for. If the |str| contains exactly one type of
+// notification, then this type is returned, otherwise a null QVariant is returned.
+QVariant getNotificationTypeFromString(const QString &str) {
+    using Notification = U2::GTUtilsDashboard::Notification;
+
+    QVariant returnable;
+    constexpr int typesNumber = sizeof(Notification::typesNames) / sizeof(QString);
+    int counter = 0;
+    Notification::Type type;
+
+    for (int typeInd = 0; typeInd < typesNumber; ++typeInd) {
+        if (str.contains(Notification::typesNames[typeInd], Qt::CaseInsensitive)) {
+            ++counter;
+            type = static_cast<Notification::Type>(typeInd);
+        }
+    }
+
+    if (counter == 1) {
+        returnable = static_cast<int>(type);
+    }
+    return returnable;
+};
+
+#define GT_CLASS_NAME "GTUtilsDashboard"
+
+#define GT_METHOD_NAME "getCellText"
+// Returns text from the (row, column) table cell. The (row, column) cell must exist
+QString getNotificationCellText(HI::GUITestOpStatus &os, const QGridLayout &tableLayout, const int row, const int col) {
+    QString text;
+    const QWidget *const cellWidget = tableLayout.itemAtPosition(row, col)->widget();
+
+    if (cellWidget != nullptr && cellWidget->objectName() == "tableCell") {
+        if (const QLayout *const cellLayout = cellWidget->layout()) {
+            for (int labelInd = 0; labelInd < cellLayout->count(); ++labelInd) {
+                if (const auto *const label = qobject_cast<QLabel *>(cellLayout->itemAt(labelInd)->widget())) {
+                    text = label->text();
+                }
+            }
+        }
+    }
+    return text;
```

Comment:
Normally you should fail in the test method if the result is not found/available by default.